Mobile/app/src/main/java/com/example/mobile/MainActivity.kt

```diff
@@ -1,11 +1,39 @@
 package com.example.mobile
 
+import ApiCall
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import com.example.mobile.DTOs.UserRegisterRequestDTO
+import com.example.mobile.DTOs.UserToBeStoredDTO
+import com.example.mobile.api.Auth
+import com.google.gson.Gson
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
 
 class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        val apiButton = findViewById<android.widget.Button>(R.id.buttonApi);
+        val apiCall = ApiCall()
+        val gson = Gson()
+        // example call with callbacks for register function
+        apiButton.setOnClickListener {
+            println("Button clicked")
+            val testDto = UserRegisterRequestDTO("test", "test", "test1", "test", "test", arrayListOf<String>("test","test"), "test", "test")
+            apiCall.registerUserAsync(testDto) { result, error ->
+                if (result != null) {
+                    val userToBeStored = gson.fromJson(result, UserToBeStoredDTO::class.java)
+                    println(userToBeStored)
+                } else {
+                    if (error != null) {
```

Comment:
nu putem incorpora asta in try catch in loc sa facem cu if si else?

Reply:
Practic functia asta are 2 parametri care sunt setati in functia registerUserAsync. Acolo am un try catch, daca am eroare trimit (null, error), daca nu am eroare trimit (result, null)

Reply:
okk e super atunci